core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -589,7 +593,10 @@ public void updateRetainedTransports(Collection<EquivalentAddressGroup> addrs) {
     @Override
     public ClientTransport getTransport(final EquivalentAddressGroup addressGroup) {
       checkNotNull(addressGroup, "addressGroup");
-      TransportSet ts;
+      TransportSet ts = transports.get(addressGroup);
```

Comment:
Let me see if I understand correctly: this avoids the lock in `transports`.  If the Channel is in the process of shutting down but hasn't cleared `transports`, this will grab a transport.  Later, when each transport is being shutdown, the ClientCall will try to invoke a method on the transport and get back some sort of error, safely.  

Does that sound right?  Will this PR just unmask some internal locking inside of the transports?